Add new sections to Rancher Security Best Practices (#1868)
rancher/rancher-docs: docs/reference-guides/rancher-security/rancher-security-best-practices.md

- **RKE and RKE2:** Port 6443, used by the Kubernetes API, and port 9345, used for node registration. 

These ports have TLS SAN certificates which list nodes' public IP addresses. An attacker could use that information to gain unauthorized access or monitor activity on the cluster. Protecting these ports helps mitigate against nodes' public IP addresses being disclosed to potential attackers.

> ## Rancher Username Policy

Reviewer: I believe that in this section we could highlight 2 points:
1. That `admin` is the default username of Rancher's administrator and link to our [docs](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/resources/bootstrap-password).
2. Mention that Rancher only enforces a simple [minimum password length policy](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/manage-users-and-groups#minimum-password-length).

These links are just for the sake of completeness.

By default, Kubernetes does not provide enforcement mechanisms for baseline username policies. In Rancher, this means that any enforcement of username formats, naming conventions, or baseline policies is expected to be handled by the external identity provider's policies, if such policies are in place.

Examples of potential baseline policies include:
- Requiring usernames to follow an organizational convention (e.g., `firstname.lastname`)
- Enforcing minimum or maximum length requirements
- Disallowing certain special characters
- Preventing impersonation by disallowing reserved names (e.g., `admin`, `root`)

Without these controls at the identity provider layer, there is a risk of inconsistent or insecure username practices, which can complicate access audits and lead to privilege escalation attempts.

**Recommendation:**

> We strongly recommend that customers:

Reviewer: Perhaps `recommend` could be changed to `advice`, just because we already used `**Recommendation:**` in the line above.

- [OWASP Identity Management](https://owasp.org/www-project-top-ten/2017/A2_2017-Broken_Authentication)

## WAF Rules

Rancher is designed to support a wide range of deployment scenarios, including environments where customers may programmatically automate the creation or provisioning of large numbers of clusters. Imposing strict application-level limits within Rancher itself could interfere with legitimate workloads that require dynamic scaling.

For example:
- CI/CD pipelines may create and tear down clusters frequently.
- Self-service portals could provision clusters on-demand for developers.
- Test environments may generate high volumes of API calls.

**Risk:**
Without appropriate rate limiting, adversaries could exploit unauthenticated or authenticated endpoints to:
- Exhaust resources (Denial of Service).
- Inflate storage costs.
- Degrade performance for legitimate users.

**Recommendation:**
The most effective way to mitigate this risk is to implement rate limiting and abuse protection at the infrastructure or Web Application Firewall (WAF) layer. This approach allows thresholds to be tuned for each environment's expected usage and scaling characteristics.

> 
> We recommend:

Reviewer: ```suggestion
```

Reviewer: Maybe we can remove it to avoid being redundant with the `**Recommendation:**` above.

- Configuring a Web Application Firewall or API Gateway to enforce rate limits on sensitive operations, such as cluster creation and provisioning.
- Defining thresholds based on baseline workload expectations (e.g., max requests per minute per client).
- Monitoring logs and alerting on anomalies to detect potential abuse.

> 

Reviewer: I feel that we could also add a bullet point mentioning about [project resources quotas](https://ranchermanager.docs.rancher.com/how-to-guides/advanced-user-guides/manage-projects/manage-project-resource-quotas).

- **RKE and RKE2:** Port 6443, used by the Kubernetes API, and port 9345, used for node registration. 

These ports have TLS SAN certificates which list nodes' public IP addresses. An attacker could use that information to gain unauthorized access or monitor activity on the cluster. Protecting these ports helps mitigate against nodes' public IP addresses being disclosed to potential attackers.

## Rancher Username Policy

By default, Kubernetes does not provide enforcement mechanisms for baseline username policies. In Rancher, this means that any enforcement of username formats, naming conventions, or baseline policies is expected to be handled by the external identity provider's policies, if such policies are in place.

> In Rancher, `admin` is the default username for the Administrator user, as highlighted [here](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/resources/bootstrap-password)

Reviewer 2: ```suggestion
In Rancher, `admin` is the default username for the Administrator user, as highlighted [here](../../getting-started/installation-and-upgrade/resources/bootstrap-password.md)
```

> Examples of potential baseline policies include:

Reviewer 2: ```suggestion
Examples of potential baseline policies include:

```

- Requiring usernames to follow an organizational convention (e.g., `firstname.lastname`)
- Enforcing minimum or maximum length requirements
- Disallowing certain special characters
- Preventing impersonation by disallowing reserved names (e.g., `admin`, `root`)

Without these controls at the identity provider layer, there is a risk of inconsistent or insecure username practices, which can complicate access audits and lead to privilege escalation attempts.

> Important: Rancher currently enforces only a [minimum password length](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/manage-users-and-groups#minimum-password-length)

Reviewer 2: ```suggestion

:::note Important

Rancher currently enforces only a [minimum password length](../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/manage-users-and-groups.md#minimum-password-length).

:::
```

**Recommendation:**

> We strongly advice that customers:

Reviewer 2: ```suggestion
We strongly advice that customers:

```

- Review and configure username baseline policies directly in their external identity providers (e.g., LDAP, Active Directory, SAML, or OIDC).
- Ensure that those policies align with the organization’s security and compliance requirements.
- Regularly audit user accounts to detect naming inconsistencies or policy violations.

> For more information, see:

Reviewer 2: ```suggestion
For more information, see:

```

- [OWASP Cheat Sheet: Authentication](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html)
- [OWASP Identity Management](https://owasp.org/www-project-top-ten/2017/A2_2017-Broken_Authentication)

## WAF Rules

Rancher is designed to support a wide range of deployment scenarios, including environments where customers may programmatically automate the creation or provisioning of large numbers of clusters. Imposing strict application-level limits within Rancher itself could interfere with legitimate workloads that require dynamic scaling.

> For example:

Reviewer 2: ```suggestion
For example:

```

- CI/CD pipelines may create and tear down clusters frequently.
- Self-service portals could provision clusters on-demand for developers.
- Test environments may generate high volumes of API calls.

**Risk:**

> Without appropriate rate limiting, adversaries could exploit unauthenticated or authenticated endpoints to:

Reviewer 2: ```suggestion
Without appropriate rate limiting, adversaries could exploit unauthenticated or authenticated endpoints to:

```

- Exhaust resources (Denial of Service).
- Inflate storage costs.
- Degrade performance for legitimate users.

**Recommendation:**
The most effective way to mitigate this risk is to implement rate limiting and abuse protection at the infrastructure or Web Application Firewall (WAF) layer. This approach allows thresholds to be tuned for each environment's expected usage and scaling characteristics. Some examples of controls can be:

- Configuring a Web Application Firewall or API Gateway to enforce rate limits on sensitive operations, such as cluster creation and provisioning.
- Defining thresholds based on baseline workload expectations (e.g., max requests per minute per client).
- Monitoring logs and alerting on anomalies to detect potential abuse.
- Apply a resource quota, which is a Rancher feature that limits the resources available to a project or namespace.

> For more information, see:

Reviewer 2: ```suggestion
For more information, see:

```

> - [Project Resource Quotas](https://ranchermanager.docs.rancher.com/how-to-guides/advanced-user-guides/manage-projects/manage-project-resource-quotas)

Reviewer 2: ```suggestion
- [Project Resource Quotas](../../how-to-guides/advanced-user-guides/manage-projects/manage-project-resource-quotas/manage-project-resource-quotas.md)
```
